Add intermediate snapshot logic for cloning a volume

sbesson opened the pull request:
Recent Embassy policies have impacted our workflow to clone volumes from existing running volumes as the writing speed (200Mbs) is now hitting timeouts for our TB scale volumes.

This PR creates an intermediate temporary snapshot that is used as the source of creation of the new volume leading to response times similar to the ones that have been used previously.

joshmoore commented on tasks/main.yml:
```diff
@@ -1,15 +1,31 @@
 ---
 # tasks file for roles/openstack-volume-storage
 
+- name: openstack volume | create snapshot
```
Would this be a use for a `block` with a clause to delete regardless?

joshmoore commented:
Lovely!

sbesson commented:
The last commits were successfully used to create a pilot, creating and deleting a temporary snapshot without hitting any timeout.

<details>
    <summary>Extract of provisioning playbook execution</summary>
```
TASK [ome.openstack_volume_storage : openstack volume | create temporary snapshot name] **************************************************************************************************
ok: [localhost] => {"ansible_facts": {"openstack_volume_snapshot_name": "prod103-omeroreadwrite-data-2021-12-17T12:15:17Z"}, "changed": false}

TASK [ome.openstack_volume_storage : openstack volume | create snapshot] *****************************************************************************************************************
changed: [localhost] => {"changed": true, "snapshot": {"attachments": [], "availability_zone": null, "bootable": true, "can_multiattach": false, "consistencygroup_id": null, "created_at": "2021-12-17T12:16:17.000000", "description": null, "display_description": null, "display_name": "prod103-omeroreadwrite-data-2021-12-17T12:15:17Z", "encrypted": false, "host": null, "id": "71d4a879-c2d9-4431-b288-0cc844adb4d8", "is_bootable": true, "is_encrypted": false, "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "metadata": {}, "migration_status": null, "multiattach": false, "name": "prod103-omeroreadwrite-data-2021-12-17T12:15:17Z", "os-extended-snapshot-attributes:progress": "100%", "os-extended-snapshot-attributes:project_id": "ecb43df152dc4e8bbbba2896fac99f51", "properties": {"os-extended-snapshot-attributes:progress": "100%", "os-extended-snapshot-attributes:project_id": "ecb43df152dc4e8bbbba2896fac99f51", "volume_id": "861cd59e-acb1-4da6-b6be-9fb44d14c681"}, "replication_driver": null, "replication_extended_status": null, "replication_status": null, "size": 1200, "snapshot_id": null, "source_volume_id": null, "status": "available", "updated_at": "2021-12-17T12:16:18.000000", "volume_id": "861cd59e-acb1-4da6-b6be-9fb44d14c681", "volume_type": null}}

TASK [ome.openstack_volume_storage : openstack volume | create volume] *******************************************************************************************************************
changed: [localhost] => {"changed": true, "id": "33a3a310-4efd-4485-ad6b-527cbdeab435", "volume": {"attachments": [], "availability_zone": "nova", "bootable": false, "can_multiattach": false, "consistencygroup_id": null, "created_at": "2021-12-17T12:16:21.000000", "description": null, "display_description": null, "display_name": "pilot-idr0119-omeroreadwrite-data", "encrypted": false, "host": null, "id": "33a3a310-4efd-4485-ad6b-527cbdeab435", "is_bootable": false, "is_encrypted": false, "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": "nova"}, "metadata": {}, "migration_status": null, "multiattach": false, "name": "pilot-idr0119-omeroreadwrite-data", "os-vol-tenant-attr:tenant_id": "ecb43df152dc4e8bbbba2896fac99f51", "properties": {"os-vol-tenant-attr:tenant_id": "ecb43df152dc4e8bbbba2896fac99f51", "source_volid": null, "user_id": "9e36b6c938b6a1456f420f8bf2e10ac12a2a49df4a55bd3210ef79fc112a1af7"}, "replication_driver": null, "replication_extended_status": null, "replication_status": null, "size": 1200, "snapshot_id": null, "source_volid": null, "source_volume_id": null, "status": "available", "updated_at": "2021-12-17T12:16:21.000000", "user_id": "9e36b6c938b6a1456f420f8bf2e10ac12a2a49df4a55bd3210ef79fc112a1af7", "volume_type": "__DEFAULT__"}}

TASK [ome.openstack_volume_storage : openstack volume | attach volume to host] ***********************************************************************************************************
changed: [localhost] => {"attachments": [{"attached_at": "2021-12-17T12:16:28.000000", "attachment_id": "e460b7ec-40e3-4587-a8c4-ef543d4a6afc", "device": "/dev/vdc", "host_name": "uk1-za-compute-25", "id": "33a3a310-4efd-4485-ad6b-527cbdeab435", "server_id": "7558f940-4dea-4167-ba41-820c39eb5e6d", "volume_id": "33a3a310-4efd-4485-ad6b-527cbdeab435"}], "changed": true, "id": "33a3a310-4efd-4485-ad6b-527cbdeab435", "openstack": {"OS-DCF:diskConfig": "MANUAL", "OS-EXT-AZ:availability_zone": "nova", "OS-EXT-SRV-ATTR:host": null, "OS-EXT-SRV-ATTR:hostname": null, "OS-EXT-SRV-ATTR:hypervisor_hostname": null, "OS-EXT-SRV-ATTR:instance_name": null, "OS-EXT-SRV-ATTR:kernel_id": null, "OS-EXT-SRV-ATTR:launch_index": null, "OS-EXT-SRV-ATTR:ramdisk_id": null, "OS-EXT-SRV-ATTR:reservation_id": null, "OS-EXT-SRV-ATTR:root_device_name": null, "OS-EXT-SRV-ATTR:user_data": null, "OS-EXT-STS:power_state": 1, "OS-EXT-STS:task_state": null, "OS-EXT-STS:vm_state": "active", "OS-SCH-HNT:scheduler_hints": null, "OS-SRV-USG:launched_at": "2021-12-17T12:15:52.000000", "OS-SRV-USG:terminated_at": null, "accessIPv4": "", "accessIPv6": "", "addresses": {"nfs-data-2599": [{"OS-EXT-IPS-MAC:mac_addr": "fa:16:3e:b0:9a:65", "OS-EXT-IPS:type": "fixed", "addr": "10.35.199.95", "version": 4}], "pilot": [{"OS-EXT-IPS-MAC:mac_addr": "fa:16:3e:8c:19:d2", "OS-EXT-IPS:type": "fixed", "addr": "192.168.10.51", "version": 4}]}, "adminPass": null, "az": "nova", "block_device_mapping": null, "cloud": "ebi", "config_drive": "", "created": "2021-12-17T12:15:26Z", "created_at": "2021-12-17T12:15:26Z", "description": "pilot-idr0119-omeroreadwrite", "disk_config": "MANUAL", "flavor": {"disk": 100, "ephemeral": 0, "extra_specs": {}, "name": "16c64m100d", "original_name": "16c64m100d", "ram": 65536, "swap": 0, "vcpus": 16}, "has_config_drive": false, "host": null, "hostId": "bebe51aa57f9eb6e56afa51dfc66273d49ca1d8ca236fa475638c65a", "host_id": "bebe51aa57f9eb6e56afa51dfc66273d49ca1d8ca236fa475638c65a", "host_status": null, "hostname": null, "hypervisor_hostname": null, "id": "7558f940-4dea-4167-ba41-820c39eb5e6d", "image": {"id": null}, "instance_name": null, "interface_ip": "", "kernel_id": null, "key_name": "idr-key-20210310", "launch_index": null, "launched_at": "2021-12-17T12:15:52.000000", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": "nova"}, "locked": false, "max_count": null, "metadata": {"groups": "database-hosts,pilot-idr0119-database-hosts,pilot-idr0119-hosts,omeroreadwrite-hosts,pilot-idr0119-omeroreadwrite-hosts,omero-hosts,pilot-idr0119-omero-hosts,pilot-idr0119-hosts,ebi-staging-hosts", "groups2": "pilot-idr0119-ebi-nfs,ebi-nfs,pilot-pilotidr-hosts,pilot-pilotomero-hosts,pilot-pilotdatabase-hosts", "hostname": "pilot-idr0119-omeroreadwrite", "network_order": "pilot,nfs-data-2599", "ssh_proxy_host": "required"}, "min_count": null, "name": "pilot-idr0119-omeroreadwrite", "networks": {}, "os-extended-volumes:volumes_attached": [{"delete_on_termination": true, "id": "45e37f07-eaf4-45b4-b32b-d8a2d1734c36"}, {"delete_on_termination": false, "id": "22645e68-884e-4b78-af00-216d07d303db"}], "personality": null, "power_state": 1, "private_v4": "192.168.10.51", "progress": 0, "project_id": "ecb43df152dc4e8bbbba2896fac99f51", "properties": {"OS-DCF:diskConfig": "MANUAL", "OS-EXT-AZ:availability_zone": "nova", "OS-EXT-SRV-ATTR:host": null, "OS-EXT-SRV-ATTR:hostname": null, "OS-EXT-SRV-ATTR:hypervisor_hostname": null, "OS-EXT-SRV-ATTR:instance_name": null, "OS-EXT-SRV-ATTR:kernel_id": null, "OS-EXT-SRV-ATTR:launch_index": null, "OS-EXT-SRV-ATTR:ramdisk_id": null, "OS-EXT-SRV-ATTR:reservation_id": null, "OS-EXT-SRV-ATTR:root_device_name": null, "OS-EXT-SRV-ATTR:user_data": null, "OS-EXT-STS:power_state": 1, "OS-EXT-STS:task_state": null, "OS-EXT-STS:vm_state": "active", "OS-SCH-HNT:scheduler_hints": null, "OS-SRV-USG:launched_at": "2021-12-17T12:15:52.000000", "OS-SRV-USG:terminated_at": null, "host_status": null, "locked": false, "max_count": null, "min_count": null, "os-extended-volumes:volumes_attached": [{"delete_on_termination": true, "id": "45e37f07-eaf4-45b4-b32b-d8a2d1734c36"}, {"delete_on_termination": false, "id": "22645e68-884e-4b78-af00-216d07d303db"}], "trusted_image_certificates": null}, "public_v4": "", "public_v6": "", "ramdisk_id": null, "region": "RegionOne", "reservation_id": null, "root_device_name": null, "scheduler_hints": null, "security_groups": [{"description": "Default security group", "id": "f16e77b7-f71f-425f-9577-b7a756330230", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "name": "default", "project_id": "ecb43df152dc4e8bbbba2896fac99f51", "properties": {}, "security_group_rules": [{"direction": "ingress", "ethertype": "IPv4", "group": {"name": "default", "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}, "id": "58d8091f-c4e8-403a-802b-18cd128880a5", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": null, "port_range_min": null, "project_id": "", "properties": {"group": {"name": "default", "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}}, "protocol": null, "remote_group_id": null, "remote_ip_prefix": null, "security_group_id": "f16e77b7-f71f-425f-9577-b7a756330230", "tenant_id": ""}, {"direction": "ingress", "ethertype": "IPv4", "group": {"name": "default", "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}, "id": "8614c00d-1e49-4ffc-a610-758b452c4cf9", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": null, "port_range_min": null, "project_id": "", "properties": {"group": {"name": "default", "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}}, "protocol": null, "remote_group_id": null, "remote_ip_prefix": null, "security_group_id": "f16e77b7-f71f-425f-9577-b7a756330230", "tenant_id": ""}], "stateful": true, "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}, {"description": "Default security group", "id": "f16e77b7-f71f-425f-9577-b7a756330230", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "name": "default", "project_id": "ecb43df152dc4e8bbbba2896fac99f51", "properties": {}, "security_group_rules": [{"direction": "ingress", "ethertype": "IPv4", "group": {"name": "default", "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}, "id": "58d8091f-c4e8-403a-802b-18cd128880a5", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": null, "port_range_min": null, "project_id": "", "properties": {"group": {"name": "default", "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}}, "protocol": null, "remote_group_id": null, "remote_ip_prefix": null, "security_group_id": "f16e77b7-f71f-425f-9577-b7a756330230", "tenant_id": ""}, {"direction": "ingress", "ethertype": "IPv4", "group": {"name": "default", "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}, "id": "8614c00d-1e49-4ffc-a610-758b452c4cf9", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": null, "port_range_min": null, "project_id": "", "properties": {"group": {"name": "default", "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}}, "protocol": null, "remote_group_id": null, "remote_ip_prefix": null, "security_group_id": "f16e77b7-f71f-425f-9577-b7a756330230", "tenant_id": ""}], "stateful": true, "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}, {"description": "Internal access to NFS servers (managed by Ansible)", "id": "8873fd96-a4b0-4373-a935-b22248b73c83", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "name": "idr-nfs-internal", "project_id": "ecb43df152dc4e8bbbba2896fac99f51", "properties": {}, "security_group_rules": [{"direction": "ingress", "ethertype": "IPv4", "group": {}, "id": "213d7232-508a-4828-8047-7b4e2fc0d247", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": 65535, "port_range_min": 1, "project_id": "", "properties": {"group": {}}, "protocol": "tcp", "remote_group_id": null, "remote_ip_prefix": "10.35.199.0/24", "security_group_id": "8873fd96-a4b0-4373-a935-b22248b73c83", "tenant_id": ""}, {"direction": "ingress", "ethertype": "IPv4", "group": {}, "id": "a4d3cc6c-213d-4898-a8bc-37d401f155f7", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": 65535, "port_range_min": 1, "project_id": "", "properties": {"group": {}}, "protocol": "udp", "remote_group_id": null, "remote_ip_prefix": "10.35.199.0/24", "security_group_id": "8873fd96-a4b0-4373-a935-b22248b73c83", "tenant_id": ""}], "stateful": true, "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}, {"description": "Internal access to NFS servers (managed by Ansible)", "id": "8873fd96-a4b0-4373-a935-b22248b73c83", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "name": "idr-nfs-internal", "project_id": "ecb43df152dc4e8bbbba2896fac99f51", "properties": {}, "security_group_rules": [{"direction": "ingress", "ethertype": "IPv4", "group": {}, "id": "213d7232-508a-4828-8047-7b4e2fc0d247", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": 65535, "port_range_min": 1, "project_id": "", "properties": {"group": {}}, "protocol": "tcp", "remote_group_id": null, "remote_ip_prefix": "10.35.199.0/24", "security_group_id": "8873fd96-a4b0-4373-a935-b22248b73c83", "tenant_id": ""}, {"direction": "ingress", "ethertype": "IPv4", "group": {}, "id": "a4d3cc6c-213d-4898-a8bc-37d401f155f7", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": 65535, "port_range_min": 1, "project_id": "", "properties": {"group": {}}, "protocol": "udp", "remote_group_id": null, "remote_ip_prefix": "10.35.199.0/24", "security_group_id": "8873fd96-a4b0-4373-a935-b22248b73c83", "tenant_id": ""}], "stateful": true, "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}, {"description": "External access to OMERO servers (managed by Ansible)", "id": "d8476db4-5ff7-432f-afaf-dba471bf5ef3", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "name": "idr-omero-external", "project_id": "ecb43df152dc4e8bbbba2896fac99f51", "properties": {}, "security_group_rules": [{"direction": "ingress", "ethertype": "IPv4", "group": {}, "id": "61bc67db-8fa2-40d7-9fe7-cae811b747dc", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": 14079, "port_range_min": 14060, "project_id": "", "properties": {"group": {}}, "protocol": "tcp", "remote_group_id": null, "remote_ip_prefix": "0.0.0.0/0", "security_group_id": "d8476db4-5ff7-432f-afaf-dba471bf5ef3", "tenant_id": ""}, {"direction": "ingress", "ethertype": "IPv4", "group": {}, "id": "7f269e34-aeae-4b67-90a1-0ab202d0730a", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": 4063, "port_range_min": 4063, "project_id": "", "properties": {"group": {}}, "protocol": "tcp", "remote_group_id": null, "remote_ip_prefix": "0.0.0.0/0", "security_group_id": "d8476db4-5ff7-432f-afaf-dba471bf5ef3", "tenant_id": ""}, {"direction": "ingress", "ethertype": "IPv4", "group": {}, "id": "e73fdf76-f9ff-4c28-9224-bdccaa51354b", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": 4064, "port_range_min": 4064, "project_id": "", "properties": {"group": {}}, "protocol": "tcp", "remote_group_id": null, "remote_ip_prefix": "0.0.0.0/0", "security_group_id": "d8476db4-5ff7-432f-afaf-dba471bf5ef3", "tenant_id": ""}], "stateful": true, "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}, {"description": "External access to OMERO servers (managed by Ansible)", "id": "d8476db4-5ff7-432f-afaf-dba471bf5ef3", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "name": "idr-omero-external", "project_id": "ecb43df152dc4e8bbbba2896fac99f51", "properties": {}, "security_group_rules": [{"direction": "ingress", "ethertype": "IPv4", "group": {}, "id": "61bc67db-8fa2-40d7-9fe7-cae811b747dc", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": 14079, "port_range_min": 14060, "project_id": "", "properties": {"group": {}}, "protocol": "tcp", "remote_group_id": null, "remote_ip_prefix": "0.0.0.0/0", "security_group_id": "d8476db4-5ff7-432f-afaf-dba471bf5ef3", "tenant_id": ""}, {"direction": "ingress", "ethertype": "IPv4", "group": {}, "id": "7f269e34-aeae-4b67-90a1-0ab202d0730a", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": 4063, "port_range_min": 4063, "project_id": "", "properties": {"group": {}}, "protocol": "tcp", "remote_group_id": null, "remote_ip_prefix": "0.0.0.0/0", "security_group_id": "d8476db4-5ff7-432f-afaf-dba471bf5ef3", "tenant_id": ""}, {"direction": "ingress", "ethertype": "IPv4", "group": {}, "id": "e73fdf76-f9ff-4c28-9224-bdccaa51354b", "location": {"cloud": "ebi", "project": {"domain_id": null, "domain_name": null, "id": "ecb43df152dc4e8bbbba2896fac99f51", "name": null}, "region_name": "RegionOne", "zone": null}, "port_range_max": 4064, "port_range_min": 4064, "project_id": "", "properties": {"group": {}}, "protocol": "tcp", "remote_group_id": null, "remote_ip_prefix": "0.0.0.0/0", "security_group_id": "d8476db4-5ff7-432f-afaf-dba471bf5ef3", "tenant_id": ""}], "stateful": true, "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51"}], "server_groups": null, "status": "ACTIVE", "tags": [], "task_state": null, "tenant_id": "ecb43df152dc4e8bbbba2896fac99f51", "terminated_at": null, "trusted_image_certificates": null, "updated": "2021-12-17T12:15:52Z", "user_data": null, "user_id": "9e36b6c938b6a1456f420f8bf2e10ac12a2a49df4a55bd3210ef79fc112a1af7", "vm_state": "active", "volumes": []}}

TASK [ome.openstack_volume_storage : openstack volume | delete snapshot] *****************************************************************************************************************
changed: [localhost] => {"changed": true, "snapshot_id": "71d4a879-c2d9-4431-b288-0cc844adb4d8"}
```
</details>

sbesson commented:
But after running the deployment playbooks to completion, I see empty volumes

```
[sbesson@pilot-idr0119-omeroreadwrite ~]$ df -h
Filesystem                    Size  Used Avail Use% Mounted on
devtmpfs                       32G     0   32G   0% /dev
tmpfs                          32G   16K   32G   1% /dev/shm
tmpfs                          32G  620K   32G   1% /run
tmpfs                          32G     0   32G   0% /sys/fs/cgroup
/dev/vda1                     100G  4.4G   96G   5% /
/dev/vdb                      800G  113M  800G   1% /var/lib/pgsql
/dev/vdc                      1.2T  5.0G  1.2T   1% /data
10.35.199.252:/fg_biostudies  360T  315T   46T  88% /nfs/biostudies
10.35.199.251:/fg_bioimage    522T  426T   96T  82% /nfs/bioimage
tmpfs                         6.3G     0  6.3G   0% /run/user/11615
```

joshmoore reviewed (approved):
Congrats :+1:

sbesson commented:
4300aa8 should fix the volume creation logic. I will run more tests with the creation of pilot servers.

joshmoore commented:
https://github.com/ome/ansible-role-openstack-volume-storage/commit/4300aa8112bf0ac66582c360ea4666927569c307 certainly makes sense, though I assume before the rename you could have also used `snapshot.snapshot.id`, right?

sbesson commented:
> 4300aa8 certainly makes sense, though I assume before the rename you could have also used snapshot.snapshot.id, right?

Exactly. I incorrectly assumed the command was returning the `snapshot` object directly but it is wrapped within the standard [Ansible return value structure](https://docs.ansible.com/ansible/2.8/reference_appendices/common_return_values.html#return-values). I renamed the variable as `snapshot.snapshot.id` reads wrong to me

sbesson commented:
Successfully tested the creation of a pilot. Everything worked as expected with the exception of the snapshot deletion step which I commented. 

A quick investigation, to be confirmed by EBI VAC team is that this is expected as per the Ceph setting which creates a dependency between snapshots and derived volumes (`rbd_flatten_volume_from_snapshot` is False). If confirmed, I would propose to update the workflow to not delete the snapshot as part of this role but handle this as part of the resources cleanup. The only modification would likely be to update the volume snapshot name to use the target deployment environment rather than the source volume as the prefix so that it gets picked up by https://github.com/IDR/deployment/blob/master/scripts/os-idr-delete.py

sbesson commented:
GitHub Actions builds will require the fix included in https://github.com/ome/ome-ansible-molecule/pull/27.

The last commits implement the suggestions in https://github.com/ome/ansible-role-openstack-volume-storage/pull/4#issuecomment-1006441057 and were used to create a pilot environment
![Screenshot 2022-01-10 at 14 51 16](https://user-images.githubusercontent.com/1355463/148786386-d5a2e78f-bf85-414f-9eb1-3373aed96e99.png)

joshmoore commented on README.md:
```diff
@@ -19,6 +19,10 @@ Required variables:
 - `openstack_volume_type`: The type of volume
+- `openstack_volume_source`: The name of the volume to clone.
+
+If `openstack_volume_source` is specified, the role will create a temporary
```
Is it still "temporary" or is it "a snapshot that needs to be later deleted"?

sbesson replied:
Excellent point. Pushed f08e639 to update the README and clarify the expectation

joshmoore reviewed (approved):
One minor comment comment, otherwise :+1:
